python_modules/dagster/dagster/_core/instance/__init__.py

```diff
@@ -1682,6 +1709,49 @@ def get_materialization_count_by_partition(
     ) -> Mapping[AssetKey, Mapping[str, int]]:
         return self._event_storage.get_materialization_count_by_partition(asset_keys, after_cursor)
 
+    # partitions storage
+
+    @traced
+    def get_mutable_partitions(self, partitions_def_name: str) -> Sequence[str]:
+        check.str_param(partitions_def_name, "partitions_def_name")
+        if self._partitions_storage is None:
+            check.failed(
+                "Your storage configuration currently does not support the partitions storage."
+            )
+        return self._partitions_storage.get_partitions(partitions_def_name)
+
+    @traced
+    def add_mutable_partitions(
+        self, partitions_def_name: str, partition_keys: Sequence[str]
+    ) -> None:
+        check.str_param(partitions_def_name, "partitions_def_name")
+        check.sequence_param(partition_keys, "partition_keys", of_type=str)
+        if self._partitions_storage is None:
+            check.failed(
+                "Your storage configuration currently does not support the partitions storage."
+            )
+        return self._partitions_storage.add_partitions(partitions_def_name, partition_keys)
+
+    @traced
+    def delete_mutable_partition(self, partitions_def_name: str, partition_key: str) -> None:
+        check.str_param(partitions_def_name, "partitions_def_name")
+        check.sequence_param(partition_key, "partition_key", of_type=str)
+        if self._partitions_storage is None:
+            check.failed(
+                "Your storage configuration currently does not support the partitions storage."
+            )
+        self._partitions_storage.delete_partition(partitions_def_name, partition_key)
+
+    @traced
+    def has_mutable_partition(self, partitions_def_name: str, partition_key: str) -> bool:
```

Comment:
I suspect we'll eventually want to introduce a version of this that checks multiple partition keys. So I wonder if it's better to just do that now. I'm not super opinionated on this though.